File: backend/src/main/java/com/evalorithm/entity/Prediction.java
package com.evalorithm.entity;

import com.evalorithm.enums.RiskLevel;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "predictions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Prediction extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_profile_id", nullable = false)
    private StudentProfile studentProfile;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "subject_id", nullable = false)
    private Subject subject;

    private Double predictedMarks;

    private String predictedGrade;

    private Double passProbability;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RiskLevel riskLevel;

    @Column(columnDefinition = "TEXT")
    private String suggestedImprovement;

    private Double confidenceLevel;

    private Integer basedOnAttempts;

    @Builder.Default
    private LocalDateTime generatedAt = LocalDateTime.now();
}

File: android/app/src/main/java/com/evalorithm/data/model/Prediction.kt
package com.evalorithm.data.model

data class Prediction(
    val id: Long,
    val subjectName: String,
    val predictedMarks: Double,
    val predictedGrade: String,
    val passProbability: Double,
    val riskLevel: String,
    val suggestedImprovement: String,
    val confidenceLevel: Double,
    val generatedAt: String
)
